wpilibj/src/main/java/edu/wpi/first/wpilibj/geometry/Pose2d.java

```diff
@@ -0,0 +1,121 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2019 FIRST. All Rights Reserved.                             */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+package edu.wpi.first.wpilibj.geometry;
+
+/**
+ * Represents a 2d pose containing translational and rotational elements.
+ */
+public class Pose2d {
+  private Translation2d m_translation;
+  private Rotation2d m_rotation;
+
+  /**
+   * Constructs a pose at the origin facing toward the positive X axis.
+   * (Translation2d{0, 0} and Rotation{0})
+   */
+  public Pose2d() {
+    m_translation = new Translation2d();
+    m_rotation = new Rotation2d();
+  }
+
+  /**
+   * Constructs a pose with the specified translation and rotation.
+   */
+  public Pose2d(Translation2d translation, Rotation2d rotation) {
+    m_translation = translation;
+    m_rotation = rotation;
+  }
+
+  /**
+   * Convenience constructors that takes in x and y values directly instead of
+   * having to construct a Translation2d.
+   */
+  @SuppressWarnings("ParameterName")
+  public Pose2d(double x, double y, Rotation2d rotation) {
```

Comment:
Should rotation be added as well?

Reply:
What do you mean?

Reply:
I think he means a separate overload for rotation as a double in radians.

Reply:
I think it'd be better to make rotation a double here. Adding a separate overload with just rotation might be getting unnecessary

Reply:
What do others think? Personally, I feel that users would use this constructor in its current form a lot more because people are more inclined to degrees. Making this a double would mean that users would have input angles in radians.

Reply:
Since Rotation2d has a static helper for going from radians or degrees to a Rotation2d, I don't see why this shouldn't just take Rotation2d. It's more expressive than just throwing a magic number in there.